pool/boltdb.go

```diff
@@ -319,7 +228,40 @@ func nanoToBigEndianBytes(nano int64) []byte {
 	return b
 }
 
-func (db *BoltDB) persistPoolMode(mode uint32) error {
+func (db *BoltDB) fetchSoloPoolMode() (bool, error) {
+	// SoloPoolMode is stored as a uint32 for historical reasons.
```

Comment:
Probably need to work on this again at some point. Currently not being particularly efficient with the data type being used to store a bool but it works.

Reply:
Just realized postgres support will allow additional pool modes (for example a pool instance that is part of a cluster) so we can tackle this when that is implemented.

Reply:
Using a boolean is not a good idea if we expect pool mode to have >2 values in future. I think its better to revert back to a number, but rather than using uint32 we should have named with specific values, rather than a naked `0/1`. Maybe like this?

```
type PoolMode uint32

const (
	SoloPool   PoolMode = 0
	SharedPool PoolMode = 1
)

func setPoolMode(PoolMode)

func fetchPoolMode() (PoolMode, error)
```

Reply:
I was thinking of a `mode` field with options like `solo`, `public`, `cluster`. Its more descriptive than using number values.